main.py

```diff
@@ -76,7 +76,8 @@ def clear_postcode():
 
 @app.route('/candidates')
 def candidates():
-    if 'postcode' not in flask.session:
+    if 'postcode' not in flask.session or 'constituency' not in flask.session:
+        flask.flash("Something went wrong", 'danger')
```

Comment:
This isn't actually a flash error - it's right behaviour if they bookmark the `/candidates` URL and then go to it without cookies.

Checking constituency too is good idea - but can remove flash.